test/jdk/javax/swing/JComponent/7154030/bug7154030.java

```diff
@@ -81,25 +84,35 @@ public void run() {
                     button.setOpaque(true);
                     button.setVisible(false);
                     desktop.add(button);
+                    desktop.setMinimumSize(new Dimension(300, 300));
+                    desktop.setMaximumSize(new Dimension(300, 300));
 
                     frame.setContentPane(desktop);
-                    frame.setSize(300, 300);
+                    frame.setMinimumSize(new Dimension(350, 350));
+                    frame.setMaximumSize(new Dimension(350, 350));
+                    frame.pack();
                     frame.setLocationRelativeTo(null);
                     frame.setVisible(true);
                     frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
                 }
             });
 
-            robot.delay(1000);
             robot.waitForIdle(1000);
+            robot.delay(1000);
 
+            Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+            Rectangle screen = new Rectangle(0, 0, (int) screenSize.getWidth(), (int) screenSize.getHeight());
             Rectangle bounds = frame.getBounds();
             Insets insets = frame.getInsets();
             locx = bounds.x + insets.left;
             locy = bounds.y + insets.top;
             frw = bounds.width - insets.left - insets.right;
             frh = bounds.height - insets.top - insets.bottom;
 
+            BufferedImage fullScreen = robot.createScreenCapture(screen);
+            Graphics g = fullScreen.getGraphics();
+            g.setColor(Color.RED);
+            g.drawRect(locx - 1, locy - 1, frw + 1, frh + 1);
```

Comment:
I believe we need to call g.dispose to dispose this object which we normallydo after getGraphics() call.

Reply:
It is not necessary in such a short test, JVM will exit after test ends which will clear all the resources captured by this Graphics instance.

Reply:
BTW it will be closed/released only if the tests are run in the othervm mode, but jtreg tests might share the single VM for a group of tests.